packages/mdl-base-component/adapter.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * A basic component adapter.
+ */
+export const MDLBaseAdapter = {
+  getBoundingClientRect(id) {
+    return this.elements_[id].getBoundingClientRect();
+  },
+
+  setStyles(id, styles) {
+    Object.assign(this.elements_[id].style, styles);
+  },
+
+  forceLayout(id) {
+    return this.elements_[id].offsetWidth;
+  },
+
+  addClass(id, cls) {
+    this.elements_[id].classList.add(cls);
+  },
+
+  removeClass(id, cls) {
+    this.elements_[id].classList.remove(cls);
+  },
+
+  get(...symbolPath) {
```

Comment:
Travis and I discussed this and we'll need to think this through further for alternatives.

Explicit methods:
Pros:
- More explicit, which is usually better, also for type systems
- Simple, straightforward
Cons:
- Code duplication
- How to share adapter code?

ID/ref
Pros:
- Cuts down on code duplication for ripple, or components that have require multiple operations over a set of components
Cons:
- In theory, IDs are unnecessary bloat (need to create string for every obj to be referenced)
- Name overlapping for generic terms (layout, get, etc.)
- why/how choose a ref function